torch/nn/functional.py

```diff
@@ -337,18 +338,36 @@ def fractional_max_pool2d(input, kernel_size, output_size=None,
     """
     if output_size is None and output_ratio is None:
         raise ValueError("fractional_max_pool2d requires specifying either "
-                         "an output_size, or a output_ratio")
+                         "an output_size or an output_ratio")
     if output_size is None:
-        _output_ratio = _pair(output_ratio)
-        _output_size = (int(input.size(2) * _output_ratio[0]),
-                        int(input.size(3) * _output_ratio[1]))
+        _output_ratio = _pair(torch.jit._unwrap_optional(output_ratio))
+        _output_size = [int(input.size(2) * _output_ratio[0]),
+                        int(input.size(3) * _output_ratio[1])]
     else:
         _output_size = torch.jit._unwrap_optional(output_size)
 
     if _random_samples is None:
-        _random_samples = input.new(input.size(0), input.size(1), 2).uniform_()
-    ret = torch._C._nn.fractional_max_pool2d(input, kernel_size, _output_size, _random_samples)
-    return ret if return_indices else ret[0]
+        _random_samples = torch.rand(input.size(0), input.size(1), 2, dtype=input.dtype)
```

Comment:
We don't have support for `new`, but `rand` pulls from the same uniform distribution so this change should be equivalent

Reply:
Do you also need to set the device as well ?  `By default, the returned Tensor has the same torch.dtype and torch.device as this tensor. `